src/app/modules/item/services/item-task-answer.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { combineLatest, EMPTY, forkJoin, interval, Observable, of, Subject } from 'rxjs';
+import { catchError, distinctUntilChanged, mapTo, shareReplay, skip, switchMap, withLatestFrom } from 'rxjs/operators';
+import { SECONDS } from 'src/app/shared/helpers/duration';
+import { repeatLatestWhen } from 'src/app/shared/helpers/repeatLatestWhen';
+import { AnswerTokenService } from '../http-services/answer-token.service';
+import { Answer, GetCurrentAnswerService } from '../http-services/get-current-answer.service';
+import { GradeService } from '../http-services/grade.service';
+import { UpdateCurrentAnswerService } from '../http-services/update-current-answer.service';
+import { ItemTaskInitService } from './item-task-init.service';
+
+const answerAndStateSaveInterval = 1*SECONDS;
+
+@Injectable()
+export class ItemTaskAnswerService implements OnDestroy {
+  private task$ = this.taskInitService.task$;
+  private config$ = this.taskInitService.config$;
+  private taskToken$ = this.taskInitService.taskToken$;
+
+  private currentAnswer$: Observable<Answer | null> = this.config$.pipe(
+    switchMap(({ route, attemptId }) => this.getCurrentAnswerService.get(route.id, attemptId).pipe(
+      catchError(() => this.task$.pipe(
+        switchMap(task => combineLatest([ task.getAnswer(), task.getState() ])),
+        switchMap(([ answer, state ]) => this.updateCurrentAnswerService.update(route.id, attemptId, { answer, state })),
+        switchMap(() => this.getCurrentAnswerService.get(route.id, attemptId)),
+      )),
+    )),
+    shareReplay(1), // avoid duplicate xhr calls on multiple subscriptions.
+  );
+
+  private reloadAnswerAndStateSubject = new Subject<void>();
+  private reloadAnswerAndState$ = combineLatest([ this.currentAnswer$, this.task$ ]).pipe(
```

Comment:
Technically to me the naming reflects what the observable does: reloading answer and state, then completing.
I don't really see a clearer way to express that, is a problem of using `reload` instead of `reloaded`, or something like that ?

I can split them into `reloadedAnswer$` and `reloadedState$`, I suppose it's more declarative this way and reflects actually a state

Reply:
fixed, I renamed it to `reloadedAnswerAndState$`

Reply:
(without checking the details/reasons to do it) I still feel there is something strange when I see an observable called `reloadedAnswerAndState$`... while observable are used to follow reloaded elements... so we would expect to observe an "observable<answer,state>" if I want to follow "reloaded answer and state". 
I suppose there is a reason to do this, I'll recheck later.